src/Swift.Bindings/src/Marshaler/BoundGenericsHandler.cs

```diff
@@ -0,0 +1,146 @@
+namespace BindingsGeneration;
+
+/// <summary>
+/// Helper class for handling bound generic types in Swift bindings.
+/// It translates Swift generic types into their C# representations and provides
+/// type information for marshalling.
+/// </summary>
+public class BoundGenericsHandler
+{
+    private readonly ITypeDatabase _typeDatabase;
+
+    public BoundGenericsHandler(ITypeDatabase typeDatabase)
+    {
+        _typeDatabase = typeDatabase;
+    }
+
+    // Almost all generics will be projected into C# as classes.
+    // This collection contains types that must be marshalled as structs.
+    // We might introduce a new field in the TypeRecord to indicate this
+    private static readonly HashSet<SwiftTypeName> s_structGenerics = new()
+        {
+            SwiftTypeName.FromModuleQualifiedName("Swift.UnsafeMutableBufferPointer"),
+            SwiftTypeName.FromModuleQualifiedName("Swift.UnsafeMutablePointer"),
+        };
+
+    // TODO: Add more types as needed.
+    // Mapping of Swift generic types to their corresponding buffer types.
+    private static readonly Dictionary<SwiftTypeName, string> s_bufferTypeMap = new()
+        {
+            { SwiftTypeName.FromModuleQualifiedName("Swift.Array"), "Swift.ArrayBuffer" },
+            { SwiftTypeName.FromModuleQualifiedName("Swift.Set"), "Swift.Variant" }
+        };
+
+    /// <summary>
+    /// Determines whether the specified property declaration represents a bound generic type.
+    /// </summary>
+    /// <param name="propertyDecl">The property declaration.</param>
+    /// <returns><c>true</c> if the property’s Swift type contains generic parameters; otherwise, <c>false</c>.</returns>
+    public bool IsBoundGeneric(PropertyDecl propertyDecl) =>
+        propertyDecl.SwiftTypeSpec is NamedTypeSpec namedTypeSpec && namedTypeSpec.ContainsGenericParameters; // TODO: Check whether return type is not type's generic parameter
```

Comment:
We should start getting into habit of not adding TODOs. If there's a piece of implementation we want to postpone - create an issue and link it here. TODO alone will never get fixed :-)

Reply:
Right, https://github.com/dotnet/runtimelab/issues/3013. Added to the code.